### PR TITLE
Add  RestorationManager disposals in test/services/restoration_test.dart.

### LINKED ISSUE #134831: Restauration manager not disposed
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. replace `testWidgets` in tests/services/RestaurationTests.dart with `testWidgetsWithLeakTracking`
2. Run tests
3. See error

Or remove allowList is #134832
```dart
leakTrackingTestConfig: const LeakTrackingTestConfig(notDisposedAllowList:
      <String, int?>{'RestorationManager': 1})
```

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR adds `RestorationManager` objects disposal in `test/services/restoration_test.dart`.

### Description
- Fixes https://github.com/flutter/flutter/issues/134831.

### Tests
- Updates `test/services/restoration_test.dart` to use `testWidgetsWithLeakTracking`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (polina-c) [inline packages/flutter/.vscode/settings.json:1]: Can you remove this?

**Review by polina-c — CHANGES_REQUESTED** [review]

File: packages/flutter/.vscode/settings.json (hunk 1 above)
Comment (ksokolovskyi) [reply packages/flutter/.vscode/settings.json]: Sorry 🤦‍♂️, fixed.

Comment (polina-c) [reply packages/flutter/.vscode/settings.json]: I think you wanted to delete your edit, but not the entire file  :)

Comment (ksokolovskyi) [reply packages/flutter/.vscode/settings.json]: It seems like now all is fine :)

**Review by polina-c — APPROVED** [review]

**Comment by polina-c** [comment]
Are there other instances of https://github.com/flutter/flutter/issues/134831 in the code?
If no, let's close the issue.

**Comment by ksokolovskyi** [comment]
@polina-c, thanks for the review.
I will take a look and write back, ok?

**Pull request merged**